packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -124,7 +124,7 @@ class ErrorHandlingFileSystem extends ForwardingFileSystem {
 
   @override
   Directory get systemTempDirectory {
-    return directory(delegate.systemTempDirectory);
+    return _runSync(() => directory(delegate.systemTempDirectory), platform: _platform);
```

Comment:
To summarize everything, `delegate.systemTempDirectory` could do anything, including making file system calls, so we need to wrap this with `_runSync` to catch any `FileSystemException` (that should result in a graceful tool exit).